File: src/tools/line.js
// @flow

import Bitmap from '../view/bitmap';
import Segment from '../math/segment';

class Line {
  color: string;
  width: number;

  constructor(color: string, width: number) {
    this.color = color;
    this.width = width;
  }

  draw(bitmap: Bitmap, segment: Segment) {
    throw new TypeError('Not Implemented');
  }

  set setColor(value: string) {
    this.color = value;
  }

  set setWidth(value: number) {
    return this.width;
  }
}

export default Line;

File: lib/algorithms/line.dart
import 'Package:GraphicsApp/view/bitmap.dart';
import 'Package:GraphicsApp/math/segment.dart';
import 'package:GraphicsApp/data/color.dart';

abstract class Line {
  Color _color;
  num _width;

  Line(this._color, this._width) {}

  void draw(Bitmap bitmap, Segment segment);

  Color get color => _color;
  void set color(color) => _color = color;

  num get width => _width;
  void set width(width) => _width = width;
}
